src/codegen/intrin.cc

```diff
@@ -0,0 +1,74 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file intrin.cc
+ * \brief Support for intrinsic registry.
+ */
+#include <tvm/base.h>
+#include <tvm/intrin.h>
+
+#include <memory>
+#include <atomic>
+#include <mutex>
+#include <unordered_set>
+
+namespace dmlc {
+// enable registry
+DMLC_REGISTRY_ENABLE(tvm::Intrin);
+}  // namespace dmlc
+
+namespace tvm {
+
+// single manager of intrinsic information.
+struct IntrinManager {
+  // mutex to avoid registration from multiple threads.
+  // recursive is needed for trigger(which calls UpdateAttrMap)
+  std::recursive_mutex mutex;
+  // global intrinsic counter
+  std::atomic<int> op_counter{0};
```

Comment:
Would it better to use unsigned type for counter ?

Reply:
I just keep the same with other place

Reply:
Oh, thanks for mention this, I were just wondering about possible overflow.